tests/subset.rs

```diff
@@ -42,24 +42,33 @@ fn test_subset<A: Adversary<Subset<NodeId>>>(
     while !network.nodes.values().all(TestNode::terminated) {
         network.step();
     }
+
     // Verify that all instances output the same set.
-    let mut expected = None;
+    let observer: BTreeSet<_> = network.observer.outputs().iter().cloned().collect();
```

Comment:
`.iter().cloned().collect()` is equivalent to `.to_vec()`, I think.
Also you removed `observer.sort()`. BTW the test appears to pass without sorting. Maybe for the future (i.e., moving to the new `proptest` framework) it would still be better to sort.

Reply:
Since it's a `BTreeSet` now, `to_vec` wouldn't work, and sorting isn't possible/necessary anymore.